File: python/01.py
from functools import reduce

def sum1():
  sum = 0

  for index in range(1000):
    if index % 3 == 0 or index % 5 == 0:
      sum += index

  return sum

print(sum1())

def sum2():
  return sum([index for index in range(1000) if index % 3 == 0 or index % 5 == 0])

print(sum2())

def sum3():
  return reduce(lambda acc,val: acc+val if val % 3 == 0 or val % 5 == 0 else acc, range(1000))

print(sum3())

File: go/01.go
package main

func sum1() int {
	sum := 0

	for i := 0; i < 1000; i++ {
		if i % 3 == 0 || i % 5 == 0 {
			sum += i
		}
	}

	return sum
}

func sum2() int {
	var sum int
	var a [1000]int

	for i := range a {
		if i % 3 == 0 || i % 5 == 0 {
			sum += i
		}
	}

	return sum
}
